### PR TITLE
CORE-3930: Added topic that can be used to set footprint padding.

### PR DESCRIPTION
**Pull request by p6chen**
@skaynama @jasonimercer

### CODE REVIEW

File: costmap_2d/src/costmap_2d_ros.cpp [hunk 1]
```diff
@@ -356,6 +360,17 @@ void Costmap2DROS::setStaticRobotFootprint(const std::vector<geometry_msgs::Poin
...
 
+void Costmap2DROS::setRobotFootprintPadding(const std_msgs::Float32& padding)
+{
+  footprint_padding_ = padding.data;
+
+  // Repad the footprint with the new padding
+  padded_footprint_ = unpadded_footprint_;
+  padFootprint(padded_footprint_, footprint_padding_);
+
+  layered_costmap_->setFootprint(padded_footprint_);
```
Comment (jasonimercer) [inline costmap_2d/src/costmap_2d_ros.cpp]: setting the footprint here will trigger an update of the inflation kernel. If it's possible for this to be called while inflation is happening then a crash could occur as the existing cache for memory is deleted, the pointer is set to NULL and then it's reallocated and recalculated.

Comment (p6chen) [inline costmap_2d/src/costmap_2d_ros.cpp]: How does this differ from setting the footprint in setUnpaddedRobotFootprint() line 347 above? Is there vulnerable to a crash as well? That function runs whenever the setFootprint topic is sent, which is analogous to the padding topic this PR adds. They both run in the same place one after another.

Comment (jasonimercer) [inline costmap_2d/src/costmap_2d_ros.cpp]: If it's running in a safe state them I'm OK with it. The setUnpaddedRobotFootprint is a good example of how this is safe.

**Comment by jasonimercer** [comment]
LGTM. In the future we could consider using cpr_spatial.

**Comment by skaynama** [comment]
So what's the function of ```footprint_padding_``` anymore if it's not used for footprint clearing?

**Comment by p6chen** [comment]
Footprint padding will still retain its legacy functionality. If you specify a footprint padding, it will still pad the costmap footprints by that amount.

**Comment by skaynama** [comment]
Sure, but that footprint doesn't do anything (since we have static inflation). That is why we added this comment in the platform navigation packages:
```
footprint_padding: -0.1  # This padding is and should only be used for footprint clearing.
```
So now essentially that variable has no worth since its only functionality, padding for footprint clearing, has been transferred to the new variable ```footprint_clearing_padding_```.

This PR is good. But I think we either need to unify the two competing paddings, or else add comment in the platform navigation packages to clarify that ```footprint_clearing``` pads the costmap footprints but its footprint clearing functionality could be overridden via the new topic pub.

**Comment by p6chen** [comment]
Yea I will fix the comment in the navigation packages. I don't think they are competing. They do very distinct functions. footprint_padding_ pads the actual footprint of move_base (it is just that we don't use it to define the inflation radius anymore). However, lets say someone other move_base compatible plugin was to use that footprint value, the padding would still affect it. Just that for now it is unused. Whereas the new parameter is specifically for adjusting the padding on how much of the padded footprint to clear away.

**Comment by skaynama** [comment]
:+1:

**Pull request merged**